app/models/user.rb

```diff
@@ -4,4 +4,9 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable, :omniauthable, :recoverable,
   devise :database_authenticatable, :registerable, :rememberable, :validatable
+  enum role: { user: 0, admin: 1 }
```

Comment:
`user.user? => false` sounds weird. I would try to rename `user` role to something else, maybe `end_user`.

I think admin is an user (because the model is called `user`). So you need another name for non-admin role.